lib/vector-buffers/src/topology/channel/sender.rs

```diff
@@ -304,7 +304,7 @@ impl<T: Bufferable> Sink<T> for BufferSender<T> {
         let this = self.project();
         let item_size = this.instrumentation.as_ref().map(|_| item.size_of());
 
-        let result = match this.state {
+        match mem::replace(this.state, SendState::Idle) {
```

Comment:
Nice, I didn't think about `mem::replace`.

Reply:
I feel like I end up using it almost every single time I write a FSM in Rust, haha.